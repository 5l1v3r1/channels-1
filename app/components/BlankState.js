// @flow
import React from 'react';
import styles from './BlankState.scss';

const BlankState = ({
  message
}: {
  message: string
}) => (
  <p className={styles.message}>{message}</p>
);

export default BlankState;
