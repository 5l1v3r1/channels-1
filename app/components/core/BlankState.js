// @flow
import React from 'react';
import { IconInfo } from './Icons';
import styles from './BlankState.scss';

const BlankState = ({
  message,
  color
}: {
  message: string,
  color: string
}) => (
  <div className={[styles.blankState, styles[color]].join(' ')}>
    <IconInfo className={styles.iconInfo} />
    <p className={styles.message}>{message}</p>
  </div>
);

export default BlankState;
