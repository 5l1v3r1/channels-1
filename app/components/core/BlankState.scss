@import '../../app.global.scss';

.blankState {
	position: absolute;
	top: 50%;
	left: 50%;
	transform: translate(-50%, -50%);
	display: flex;
	align-items: center;
	justify-content: center;
	width: 100%;
	padding: 1rem;

	&.white {
		.iconInfo {
			fill: $white;
		}

		.message {
			color: $white;
		}
	}

	&.black {
		.iconInfo {
			fill: $black;
		}
	}
}

.iconInfo {
	width: 21px;
	height: 21px;
}

.message {
	margin: 0 0 0 0.5rem;
}
