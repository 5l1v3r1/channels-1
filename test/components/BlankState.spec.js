import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import BlankState from '../../app/components/core/BlankState';

describe('BlankState', () => {
  it('should render self', () => {
    const enzymeWrapper = shallow(
      <BlankState message={'No channels found.'} />
    );
    const tree = toJson(enzymeWrapper);

    expect(tree).toMatchSnapshot();
  });
});
