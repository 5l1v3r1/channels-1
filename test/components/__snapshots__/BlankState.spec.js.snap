// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlankState should render self 1`] = `
<div
  className="blankState black"
>
  <Component
    className="iconInfo"
  />
  <p
    className="message"
  >
    No channels found.
  </p>
</div>
`;
